public/src/components/channelManagement/epicTests/tagsEditor.tsx

```diff
@@ -0,0 +1,109 @@
+import { Autocomplete } from '@material-ui/lab';
+import React, { useEffect, useCallback } from 'react';
+import { TextField } from '@material-ui/core';
+import throttle from 'lodash/throttle';
+
+interface Tag {
+  id: string;
+  name?: string;
+  section?: string;
+}
+
+interface TagEditorProps {
+  label: string;
+  disabled: boolean;
+  ids: string[];
+  onUpdate: (ids: string[]) => void;
+}
+
+/**
+ * Multiselect component for tags from CAPI.
+ * Searches for tags based on the user's input.
+ */
+export const TagsEditor: React.FC<TagEditorProps> = ({ label, disabled, ids, onUpdate }) => {
+  const [inputValue, setInputValue] = React.useState<string>('');
+  const [options, setOptions] = React.useState<Tag[]>([]);
+
+  const fetchTags = (value: string): Promise<void> =>
+    fetch(`/capi/tags?web-title=${value}&page-size=1000`)
+      .then(response => response.json())
+      .then(data => {
+        const newOptions = data.response.results.map(
+          (tag: { id: string; webTitle: string; sectionId: string }) => ({
+            id: tag.id,
+            name: tag.webTitle,
+            section: tag.sectionId != null ? tag.sectionId.toUpperCase() : '[no section]',
+          }),
+        );
+
+        newOptions.sort((a: Tag, b: Tag) => {
+          if (a.section != null && b.section != null) {
+            if (a.section > b.section) {
```

Comment:
what effect does sorting by section have here, given that we've also set `groupBy` on the `Autocomplete` component?

Reply:
The MUI autocomplete component likes its options sorted before displaying them, and doesn't like it if the options for each group are mixed up - it will output the group header multiple times in different places and complain about it in console warning messages.

Reply:
ah! 👍